crates/polars-plan/src/plans/aexpr/function_expr/mod.rs

```diff
@@ -1158,24 +1158,22 @@ impl IRFunctionExpr {
                 .with_flags(|f| f | FunctionFlags::INPUT_WILDCARD_EXPANSION),
 
             F::FoldHorizontal { returns_scalar, .. }
-            | F::ReduceHorizontal { returns_scalar, .. } => FunctionOptions::groupwise()
-                .with_flags(|mut f| {
-                    f |= FunctionFlags::INPUT_WILDCARD_EXPANSION;
-                    if *returns_scalar {
-                        f |= FunctionFlags::RETURNS_SCALAR;
-                    }
-                    f
+            | F::ReduceHorizontal { returns_scalar, .. } => match returns_scalar {
```

Comment:
This is quite a breaking change. Is `elementwise` really want we want here? I would not find it quite expected if people have non-elementwise expressions in `{cum,}_fold`s.